docs/record-encryption-quickstart/index.adoc

```diff
@@ -0,0 +1,254 @@
+:experimental:
+include::_assets/attributes.adoc[]
+
+:context: record-encryption-quickstart
+:quickstart: record-encryption
+
+[id="quickstart-{context}"]
+= Record Encryption Quickstart
+
+[role="_abstract"]
+
+You want an encryption-at-rest solution for data stored in your Apache Kafka?
+This quickstart guide will show you how to do that on Kubernetes ... from scratch ... without external dependencies.
+You'll be encrypting in a snap!
+
+== Objectives
+
+* Deploy a Kafka Cluster to a Kubernetes cluster.
+* Deploy a Key Management Service (KMS) - we'll show you how with either HashiCorp Vault or AWS Localstack.
+* Deploy Kroxylicious and configure it to proxy the cluster and apply Record Encryption.
+* Produce/consume records to Kafka via the proxy demonstrating the transparent encryption.
+* Verify that the records are encrypted on the broker.
+
+WARNING: You *really don't* want to use this quickstart to deploy an environment that will be used in *production*.
+The quickstart deliberately keeps things quick and simple... there's no authentication, no TLS, no redundancy... Development purposes only - please!.
+Refer to the documentation for Kroxylicious, Strimzi and your KMS for production best practices.
+
+== Before you begin
+
+You'll need the following a developer machine with the following software installed.
+
+* https://minikube.sigs.k8s.io/docs/start[Minikube]
+* https://helm.sh/docs/intro/install/[Helm]
+
+== Start Minikube
+
+Let's get the Kubernetes Cluster up and running.  Minikube defaults work just fine for this quickstart.
+
+[source,shell]
+----
+minikube start
+----
+
+== Install the software
+
+=== Install a KMS
+
+Record Encryption needs somewhere to store its encryption keys, so you need to provide a KMS.
+Use Helm to install either HashiCorp Vault or AWS LocalStack.
+
+==== HashiCorp Vault
+
+[source,shell]
+----
+helm repo add --force-update hashicorp https://helm.releases.hashicorp.com
+helm upgrade --install vault hashicorp/vault --namespace vault --create-namespace --version 0.30.1 --set server.dev.enabled=true,server.dev.devRootToken=myroottoken --wait
+----
+
+You also need to enable the Transit Secrets Engine in Vault.
+
+[source,shell]
+----
+kubectl exec -n vault statefulsets/vault -- vault secrets enable transit
+----
+
+==== AWS Local Stack
+
+[source,shell]
+----
+helm repo add --force-update localstack-repo https://helm.localstack.cloud
+helm upgrade --install localstack localstack-repo/localstack --namespace localstack --create-namespace --version 0.6.24 --set service.type=ClusterIP --wait
+----
+
+=== Install the Strimzi Operator
+
+We are going to use Strimzi to deploy the Kafka Cluster that will be proxied.
+Let's install Strimzi next:
+
+[source,shell]
+----
+helm repo add --force-update strimzi https://strimzi.io/charts/
```

Comment:
I've used Helm here only because we have to use Helm for Vault and Localstack.  We could switch to kubectl for Strimzi if people prefer.

Reply:
If we didn't use `kubectl` later on anyway then I'd see the benefit (few prereqs => more likely readers will actually attempt following along). But since you _do_ use `kubectl`, and since Strimzi is most commonly installed using it, I wonder what the benefit to using `helm` here is? Is it simply that we don't need commands for waiting for the cluster to come up?

Reply:
I switch to kubectl, then switched back to Helm.  If we don't use Helm, then you need to process the yaml to change namespace references from my-project to `kafka`.  If I use Helm, that's taken care of for me which keeps the commands the user runs that little bit me simple.